### ISSUE TITLE
Problems with special characters in file path

### ISSUE THREAD

**Issue by AleMax** [joaquintorres/autosubsync-mpv#18]
Hello,

I'm using Linux (Manjaro) and noticed that the subtitle syncing process fails when there are special characters inside the subtitle file name / path.
It happened both with a space and apostrophe. 
They seem to get replaced with a Percent-encoded variant, and then the script fails to find / open the file path that got percent-encoded.

A file with a space:
![subtitle_space](https://user-images.githubusercontent.com/24979735/166102144-38d23f79-8dfe-48db-baeb-9b52bd03cddd.png)

A file with an apostrophe:
![subtitle_apostrophe](https://user-images.githubusercontent.com/24979735/166102147-3a3c0cd9-48dc-483b-9e5b-96e355535959.png)

I'm sorry, i personally don't know Lua and don't really have the knowledge to debug it myself, but it definetly fails inside the file_exists(filepath) function, which then presumable gets called somewhere.
I would assume either the filepath somewhere gets a weird encoding, which isn't wanted, or if that is wanted, that the relevant Lua functions to open a file / check if a file exists are not able to work with such an encoded filepath?
I also don't know if this also occurs on Windows, as i don't use that and can't test on that.

It is not the biggest problem, as i can just remove such characters from the filenames, 
but it would be cool if this could get fixed!

Thank you

**Comment by tatsumoto-ren** [comment]
I'll try testing it with different subtitle files.

**Comment by tatsumoto-ren** [comment]
I cannot reproduce. What's the name of the subtitle file?

**Comment by AleMax** [comment]
So i decided to debug it a little bit.

I took a subtitle file with the exact name `One Week Friends.JA.srt`.
And the exact steps i took are:

1. Load the Video file with `mpv Isshuukan\ Friends.\ \[01\].mkv`.
2. Drag and drop the subtitle file from the explorer (Dolphin) into the mpv window.
3. Press 'n', select sync (doesn't matter which one)

Then i get the error
`[autosubsync] Subtitle synchronization failed:`
`[autosubsync] Couldn't find /home/alemax/test/One%20Week%20Friends.JA.srt`

I played a bit around to find out, that inside the script, inside the function `get_active_track` where the tracks are requested from mpv with `mp.get_property_native('track-list')` the file path _already_ is percent encoded, which means this is coming from mpv.
I also found a function `mp_url_escape` inside the source of mpv (`stream/stream.c`), so this is apparently in fact something mpv does. I did not actually debug it properly and can't verify if that function _actually_ gets called before the filepath is sent to the lua script, but i would assume so.


Now i also tested it out with a different set of steps:

1. Load the video _and_ subtitle with `mpv Isshuukan\ Friends.\ \[01\].mkv --sub-file="One Week Friends.JA.srt"`
2. Press 'n', select sync (doesn't matter which one)

Now the subtitle syncing process works, and the string the lua script gets inside `get_active_track` is, as expected `One Week Friends.JA.srt`, and not unlike above `One%20Week%20Friends.JA.srt`.

So the difference here lies in whether i drag and drop the subtitle file into the mpv window or starting mpv with the subtitle file directly.
I'm not entirely sure why this is the case, or if it is wanted from mpv's end. 

Now this is for you to decide if you want to try to catch this and de-encode an URL-Encoded file path, if such occurs, or if this is something mpv itself should handle different. (Or if the matter is quite specific anyway and therefore not important enough to warrant a change)
